cmd/codeqlExecuteScan_test.go

```diff
@@ -121,58 +136,75 @@ func TestGetGitRepoInfo(t *testing.T) {
-		assert.Equal(t, "", orgName)
-		assert.Equal(t, "", repoName)
-	})
-	t.Run("Invalid repository without tool instance", func(t *testing.T) {
-		repository := "/Testing/fortify"
-		toolInstance, orgName, repoName, err := parseRepositoryURL(repository)
-		assert.Error(t, err)
-		assert.ErrorContains(t, err, "Unable to parse tool instance")
-		assert.Equal(t, "", toolInstance)
-		assert.Equal(t, "", orgName)
-		assert.Equal(t, "", repoName)
-	})
-	t.Run("Empty repository", func(t *testing.T) {
-		repository := ""
-		toolInstance, orgName, repoName, err := parseRepositoryURL(repository)
-		assert.Error(t, err)
-		assert.ErrorContains(t, err, "Repository param is not set")
-		assert.Equal(t, "", toolInstance)
-		assert.Equal(t, "", orgName)
-		assert.Equal(t, "", repoName)
+
+	t.Run("Valid url with dots URL1", func(t *testing.T) {
+		config := codeqlExecuteScanOptions{Repository: "https://github.hello.test/Testing/com.sap.codeql.git", AnalyzedRef: "refs/head/branch", CommitID: "abcd1234"}
+		repoInfo := initGitInfo(&config)
+		assert.Equal(t, "abcd1234", repoInfo.commitId)
+		assert.Equal(t, "Testing", repoInfo.owner)
+		assert.Equal(t, "com.sap.codeql", repoInfo.repo)
+		assert.Equal(t, "refs/head/branch", repoInfo.ref)
+		assert.Equal(t, "https://github.hello.test", repoInfo.serverUrl)
+	})
+
+	t.Run("Valid url with dots URL2", func(t *testing.T) {
+		config := codeqlExecuteScanOptions{Repository: "https://github.hello.test/Testing/com.sap.codeql", AnalyzedRef: "refs/head/branch", CommitID: "abcd1234"}
+		repoInfo := initGitInfo(&config)
+		assert.Equal(t, "abcd1234", repoInfo.commitId)
+		assert.Equal(t, "Testing", repoInfo.owner)
+		assert.Equal(t, "com.sap.codeql", repoInfo.repo)
+		assert.Equal(t, "refs/head/branch", repoInfo.ref)
+		assert.Equal(t, "https://github.hello.test", repoInfo.serverUrl)
+	})
+
+	t.Run("Valid url with username and token URL1", func(t *testing.T) {
+		config := codeqlExecuteScanOptions{Repository: "https://username:token@github.hello.test/Testing/codeql.git", AnalyzedRef: "refs/head/branch", CommitID: "abcd1234"}
+		repoInfo := initGitInfo(&config)
+		assert.Equal(t, "abcd1234", repoInfo.commitId)
+		assert.Equal(t, "Testing", repoInfo.owner)
+		assert.Equal(t, "codeql", repoInfo.repo)
+		assert.Equal(t, "refs/head/branch", repoInfo.ref)
+		assert.Equal(t, "https://github.hello.test", repoInfo.serverUrl)
+	})
+
+	t.Run("Valid url with username and token URL2", func(t *testing.T) {
+		config := codeqlExecuteScanOptions{Repository: "https://username:token@github.hello.test/Testing/codeql", AnalyzedRef: "refs/head/branch", CommitID: "abcd1234"}
+		repoInfo := initGitInfo(&config)
+		assert.Equal(t, "abcd1234", repoInfo.commitId)
+		assert.Equal(t, "Testing", repoInfo.owner)
+		assert.Equal(t, "codeql", repoInfo.repo)
+		assert.Equal(t, "refs/head/branch", repoInfo.ref)
+		assert.Equal(t, "https://github.hello.test", repoInfo.serverUrl)
+	})
+
+	t.Run("Invalid URL with no org/reponame", func(t *testing.T) {
```

Comment:
This test case will fail if there is any orchestrator (e.g. github actions) because empty configuration options will be taken from orchestrator environment variables. this test case will only work fine locally. That's why unit tests for this PR failed